plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -364,25 +364,27 @@ public static ReaderPageSource createPageSource(
     }
 
     private static Optional<ColumnIndexStore> getColumnIndexStore(
-            Predicate parquetPredicate,
             ParquetDataSource dataSource,
             BlockMetaData blockMetadata,
             Map<List<String>, RichColumnDescriptor> descriptorsByPath,
             ParquetReaderOptions options)
     {
-        if (!options.isUseColumnIndex() || !(parquetPredicate instanceof TupleDomainParquetPredicate)) {
+        if (!options.isUseColumnIndex()) {
             return Optional.empty();
         }
 
-        return blockMetadata.getColumns().stream()
-                .filter(column -> column.getColumnIndexReference() != null && column.getOffsetIndexReference() != null)
-                .findAny()
-                .map(ignored -> TrinoColumnIndexStore.create(
-                        dataSource,
-                        blockMetadata,
-                        descriptorsByPath.keySet().stream()
-                                .map(path -> ColumnPath.get(path.toArray(String[]::new)))
-                                .collect(toImmutableSet())));
+        for (ColumnChunkMetaData column : blockMetadata.getColumns()) {
+            if (column.getColumnIndexReference() != null && column.getOffsetIndexReference() != null) {
```

Comment:
Isn't this check inverted? If any column chunk has a column index or offset index it will bail out.